### PR TITLE
Site Migration: Localize the Site Migration components

### PR DESCRIPTION
**Pull request by bisko**
#### Changes proposed in this Pull Request

* This PR adds localization to the Site Migration components 

#### Testing instructions

Unsure at the moment.

### CODE REVIEW

File: client/my-sites/migrate/components/sites-block/index.jsx [hunk 1]
```diff
@@ -94,7 +96,7 @@ export default class SitesBlock extends Component {
 				<div className="sites-block__sites-item">
 					<Site site={ targetSite } indicator={ false } />
 					<div className="sites-block__sites-labels-container">
-						<Badge type="info">This site</Badge>
+						<Badge type="info">{ translate( 'This site' ) }</Badge>
```
Comment (a8ci18n) [inline client/my-sites/migrate/components/sites-block/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **31** times:
`translate( 'This Site' )` *ES Score: 6*

File: client/my-sites/migrate/section-migrate.jsx [hunk 2]
```diff
@@ -319,26 +321,26 @@ class SectionMigrate extends Component {
...
 				/>
 				<div className="migrate__status">
-					There was an error with your import.
+					{ translate( 'There was an error with your import.' ) }
 					<br />
 					{ this.state.errorMessage }
 				</div>
 				<Button primary onClick={ this.resetMigration }>
-					Back to your site
+					{ translate( 'Back to your site' ) }
```
Comment (a8ci18n) [inline client/my-sites/migrate/section-migrate.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Back up your site' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/site-migration-add-localization)

Comment (bisko) [reply client/my-sites/migrate/section-migrate.jsx]: Different context for this one.

File: client/my-sites/migrate/section-migrate.jsx [hunk 3]
```diff
@@ -319,26 +321,26 @@ class SectionMigrate extends Component {
...
 
 	renderMigrationProgress() {
-		const { sourceSite, targetSite } = this.props;
+		const { sourceSite, targetSite, translate } = this.props;
 		const sourceSiteDomain = get( sourceSite, 'domain' );
 		const targetSiteDomain = get( targetSite, 'domain' );
 		const subHeaderText = (
 			<>
-				{ "We're moving everything from " }
+				{ translate( "We're moving everything from " ) }
```
Comment (jonathansadowski) [inline client/my-sites/migrate/section-migrate.jsx]: This won't translate well. Sentences like this should use [string substitution](https://github.com/Automattic/i18n-calypso#string-substitution).

File: client/my-sites/migrate/step-import-or-migrate.jsx [hunk 4]
```diff
@@ -47,28 +45,38 @@ class StepImportOrMigrate extends Component {
...
 			const sourceSiteDomain = get( sourceSite, 'domain' );
 			return (
 				<p>
-					You need to have{ ' ' }
-					<a href={ `https://wordpress.com/jetpack/connect/install?url=${ sourceSiteDomain }` }>
-						Jetpack
-					</a>{ ' ' }
-					installed on your site to be able to import everything.
+					{ translate(
+						"You need to have <a href='%(jetpackInstallLink)s'>Jetpack</a> installed on your site to be able to import everything.",
```
Comment (jonathansadowski) [inline client/my-sites/migrate/step-import-or-migrate.jsx]: I think the best practice here is to pass the [anchor tag as a component](https://github.com/Automattic/i18n-calypso#mixing-strings-and-markup).

File: client/my-sites/migrate/step-source-select.jsx [hunk 5]
```diff
@@ -88,18 +92,22 @@ class StepSourceSelect extends Component {
...
 				<CompactCard>
-					<CardHeading>What WordPress site do you want to import?</CardHeading>
+					<CardHeading>{ translate( 'What WordPress site do you want to import?' ) }</CardHeading>
 					<div className="migrate__explain">
-						Enter a URL and we'll help you move your site to WordPress.com. If you already have a
-						backup file, you can <a href={ uploadHref }>upload it to import content</a>.
+						{ translate(
+							"Enter a URL and we'll help you move your site to WordPress.com. If you already have a" +
+								"backup file, you can <a href='%(uploadFileLink)'>upload it to import content</a>.",
+							{ args: { uploadFileLink } }
```
Comment (jonathansadowski) [inline client/my-sites/migrate/step-source-select.jsx]: See comment about about using components.

File: client/my-sites/migrate/step-import-or-migrate.jsx (hunk 4 above)
Comment (bisko) [reply client/my-sites/migrate/step-import-or-migrate.jsx]: Thanks! Updated! Also thanks for sharing this tip!

File: client/my-sites/migrate/step-source-select.jsx (hunk 5 above)
Comment (bisko) [reply client/my-sites/migrate/step-source-select.jsx]: Thanks! Updated!

File: client/my-sites/migrate/section-migrate.jsx (hunk 3 above)
Comment (bisko) [reply client/my-sites/migrate/section-migrate.jsx]: Thanks! Updated with proper string and component substitution!

**Comment by bisko** [comment]
Updated the code to follow better practices and fixed a tiny Jetpack install URL issue.

File: client/my-sites/migrate/section-migrate.jsx [hunk 6]
```diff
@@ -287,59 +287,69 @@ class SectionMigrate extends Component {
...
-
 	renderMigrationError() {
+		const { translate } = this.props;
+
 		return (
 			<Card className="migrate__pane">
 				<FormattedHeader
 					className="migrate__section-header"
-					headerText="Import failed"
+					headerText={ translate( 'Import failed' ) }
```
Comment (a8ci18n) [inline client/my-sites/migrate/section-migrate.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **42** times:
`translate( 'Import Failed ' )` *ES Score: 9*

Comment (andfinally) [reply client/my-sites/migrate/section-migrate.jsx]: Would it work to use the suggsted string, given this is a header?

File: client/my-sites/migrate/section-migrate.jsx (hunk 3 above)
Comment (andfinally) [reply client/my-sites/migrate/section-migrate.jsx]: I wonder if Jon meant having placeholders for the `sourceSiteDomain` and `targetSiteDomain` and their wrapping elements? I think that is possible.

File: client/my-sites/migrate/step-import-or-migrate.jsx (hunk 4 above)
Comment (andfinally) [reply client/my-sites/migrate/step-import-or-migrate.jsx]: We're still mixing in markup here...

File: client/my-sites/migrate/step-source-select.jsx (hunk 5 above)
Comment (andfinally) [reply client/my-sites/migrate/step-source-select.jsx]: Same again, I think Jon means we should extract the `a` into a component.

File: client/my-sites/migrate/section-migrate.jsx (hunk 6 above)
Comment (jonathansadowski) [reply client/my-sites/migrate/section-migrate.jsx]: Honestly, the suggested string isn't great, and we should probably replace it, wherever it is. I'm not sure why the string ends in a space.

Comment (andfinally) [reply client/my-sites/migrate/section-migrate.jsx]: Ah well spotted!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/2662672

Hi @bisko, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Select a site to import into`
- `Your import has completed successfully.`
- `There was an error with your import.`
- `We're moving everything from {{sp}}%(sourceSiteDomain)s{{/sp}} to {{sp}}%(targetSiteDomain)s{{/sp}}.`
- `Import in progress`
- `A Business Plan is required to import everything.`
- `Import everything from %(sourceSiteDomain)s and overwrite everything on %(targetSiteDomain)s?`
- `All posts, pages, comments, and media`
- `All users and roles`
- `Theme, plugins, and settings`
- `Your site will keep working, but your WordPress.com dashboard will be locked during importing.`
- `Import your entire site with the Business Plan.`
- `What do you want to import?`
- `Everything`
- `All your site's content, themes, plugins, users and settings`
- `Content only`
- `Import posts, pages, comments, and media.`
- `This site is already hosted on WordPress.com`
- `This is not a WordPress site`
- `Import from WordPress`
- `What WordPress site do you want to import?`
- `Enter a URL and we'll help you move your site to WordPress.com. If you already have a WordPress export file, you can {{uploadFileLink}}upload it to import content{{/uploadFileLink}}.`
- `You need to have Jetpack installed on your site to be able to import everything. {{jetpackInstallLink}}Install Jetpack{{/jetpackInstallLink}}.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/2662672).

**Pull request merged**